### ISSUE TITLE
Improvement for NDP data sources

### ISSUE THREAD

**Issue by pengchenglai** [openforis/fra-platform#3596]
@minotogna 

**Is your feature request related to a problem? Please describe.**
As discussed with the admin, it would be beneficial to explore the possibility of implementing the data sources format of **National data** for the **NDP** (as below, removing the year and replacing the FRA variable with national classes) during the summer when summarizing for future improvements.

Thus, the proposed feature is well noted and plz put it on hold here. Thank you!

**Screenshot**
Other section:
![image](https://github.com/openforis/fra-platform/assets/99216009/acb42372-3c97-4a7e-9d51-85c3653fc919)

NDP section:
<img width="1072" height="166" alt="Image" src="https://github.com/user-attachments/assets/29122a2f-9381-43d6-9218-cb284144a725" />

**Comment by minotogna** [comment]
@pengchenglai , in the NDP page, the new DataSources table should be available after National Classes definition table ?

**Comment by pengchenglai** [comment]
@minotogna 

This feature may be implemented only after this reporting cycle. As it is not urgent, it will be discussed and decided upon along with other improvements during the summation period of summer, so nothing needs to be done and just on hold this issue for the summer discussion.

**Comment by sorja** [comment]
@minotogna Any news on this?

**Comment by minotogna** [comment]
> @minotogna Any news on this?

Nope @sorja

**Comment by pengchenglai** [comment]
@minotogna @sorja 

Let me talk with the admin and get back to you

**Comment by pengchenglai** [comment]
@minotogna shared the details with you in an email

**Comment by minotogna** [comment]
> @minotogna shared the details with you in an email

Thank you @pengchenglai ! I'll take a look next week and will let you know if anything.

**Comment by minotogna** [comment]
@pengchenglai can we migrate all past assessments as well (2020, 2025) or only latest ?

**Comment by pengchenglai** [comment]
@minotogna  You mean migrating all data sources from past assessments into the next Latest instance? 
There may be some duplication, so I am thinking we should keep only the latest. 
I will double-check with ÖJ when he is around and get back to you.

**Comment by minotogna** [comment]
> @minotogna  You mean migrating all data sources from past assessments into the next Latest instance? 
> There may be some duplication, so I am thinking we should keep only the latest. 
> I will double-check with ÖJ when he is around and get back to you.

No @pengchenglai . Data will remain in their respective cycles .

I meant only the format. Migrate the new format .

**Comment by pengchenglai** [comment]
@minotogna 

I see. This format should apply only to the latest. The layout of the past assessments needs to be aligned with the published country report.

**Comment by minotogna** [comment]
> @minotogna 
> 
> I see. This format should apply only to the latest. The layout of the past assessments needs to be aligned with the published country report.

Ok 👍 thank you @pengchenglai

**Comment by minotogna** [comment]
## TODO
@sorja @yaguzmang 

- [x] add cycle.props.ndp (+ update assessment cache)
```
ndp?: {
  dataSources: {
    version: 1 | 2
  }
}
```
```
	- fra 2020 and 2025 -> version 1
	- fra latest -> version 2
```

- [x] Database migration:
	- add column uuid (and typescript type) to original_data_point table
	- add column (optional ) section_key to descriptions table
	- migrate all odp data sources (data_source_references, data_source_methods,data_source_additional_comments) to descriptions : (reference, type, comments) -> populate section_key with odp uuid


- [x] BE fetch:
	- ensure the data sources are attached to the odp (add odp.dataSources and remove dataSourceAdditionalComments, dataSourceMethods, dataSourceReferences)

- [x] FE Render v1
	- render conditionally datasources v1 (make sure it works for 2020/2025)

- [x] FE Render v2
	- [x] Generic component 
	- [x] render v2 -> create DataSources component (pass from outside options, includeVariables, includeYear, messageTopicKey, variables (optional))

- [x] BE/FE  createOriginalDataPoint

- [x] BE/FE  updateOriginalDataPointDataSources v1

- [x] BE/FE  updateOriginalDataPointDataSources v2        

- [x] BE/FE check/update review comments v1/v2

- [x] BE/FE  removeOriginalDataPoint

- [x] BE Controller:getOriginalDataPointLastApproved
- [x] FE Data sources history v1
- [x] FE Data sources history v2

- [x] BE src/server/db/repository/assessmentCycle/repository/getMany.ts (join odp with descriptions and don't use data_source_references line 36)
- [x] BE Bulk download 
 
- [x] BE Repository: verify all queries uses new OriginalDataPointAdapter

- [x] Data Source links
        -  src/server/controller/cycleData/links/getAllLinksToVisit.ts 


- [x] Update DB fixture for tests

- [x] Swagger @ src/server/api/cycleData/originalDataPoint/getOdp.apidoc.yml
- [x] Estimation engine @ src/server/service/estimates/estimationEngine.ts
- [x] BE Verify links job

- [x] Fix NDP History description
